Sources/NIO/SocketChannel.swift

```diff
@@ -736,13 +734,17 @@ final class SocketChannel: BaseSocketChannel<Socket> {
     }
 
     init(eventLoop: SelectableEventLoop, protocolFamily: Int32) throws {
-        let socket = try Socket(protocolFamily: protocolFamily, type: Posix.SOCK_STREAM)
-        do {
-            try socket.setNonBlocking()
-        } catch let err {
-            _ = try? socket.close()
-            throw err
-        }
+        #if os(Linux)
+            let socket = try Socket(protocolFamily: protocolFamily, type: Posix.SOCK_STREAM | Linux.SOCK_NONBLOCK)
+        #else
+            let socket = try Socket(protocolFamily: protocolFamily, type: Posix.SOCK_STREAM)
+            do {
+                try socket.setNonBlocking()
+            } catch let err {
+                _ = try? socket.close()
+                throw err
+            }
```

Comment:
Can we provide a convenience class method that creates a non-blocking socket, rather than explicitly pass the type parameter? I don't love the split logic here.

Reply:
Will do, I agree, but I wanted to make minimal changes first and see how y'all felt.

Reply:
We feel great 👍 contributions highly welcome